File: src/main/java/com/github/unchama/seichiassist/data/player/AchievementPoint.kt
package com.github.unchama.seichiassist.data.player

data class AchievementPoint(val fromUnlockedAchievements: Int = 0, val used: Int = 0, val conversionCount: Int = 0) {
  val cumulativeTotal = fromUnlockedAchievements + conversionCount * 3

  val left = cumulativeTotal - used
}

File: src/main/scala/com/github/unchama/seichiassist/data/player/AchievementPoint.scala
package com.github.unchama.seichiassist.data.player

case class AchievementPoint(fromUnlockedAchievements: Int = 0, used: Int = 0, conversionCount: Int = 0) {
  val cumulativeTotal: Int = fromUnlockedAchievements + conversionCount * 3

  val left: Int = cumulativeTotal - used
}